File: LibTXNDCPy/LibTXReSciCollPy/LibTXReSciCollPy/RSCSampleFromOGWell.py
from LibTXReSciCollPy import RSCSample

class RSCSampleFromOGWell(RSCSample):
    
    """This represents a sample of some kind that comes from an oil and/or gas well.  It has properties such as the API number and operator that get integrated into the abstract field for export."""

    def __init__(self):
        super().__init__()

        self.apiNumber = None
        self.leaseName = None
        self.wellNumber = None
        self.stateName = None
        self.countyName = None
        self.countryName = None
        self.fieldName = None
        self.operatorName = None

    @property
    def apiNumber(self):
        """ API number for the well the sample comes from """
        return self.__apiNumber

    @apiNumber.setter
    def apiNumber(self, val):
        self.__apiNumber = val

    @property
    def leaseName(self):
        """ Lease name for the well the sample comes from.   Generally, combine this with the well number and you get the 'well name'"""
        return self.__leaseName

    @leaseName.setter
    def leaseName(self, val):
        self.__leaseName = val

    @property
    def wellNumber(self):
        """ Well number for the well the sample comes from.  Generally, combine this with the lease name and you get the 'well name' """
        return self.__wellNumber

    @wellNumber.setter
    def wellNumber(self, val):
        self.__wellNumber = val

    @property
    def stateName(self):
        """ The name of the state or province where the well the sample comes from is located. """
        return self.__stateName

    @stateName.setter
    def stateName(self, val):
        self.__stateName = val

    @property
    def countyName(self):
        """ The name of the county/parish where the well the sample comes from is located """
        return self.__countyName

    @countyName.setter
    def countyName(self, val):
        self.__countyName = val

    @property
    def countryName(self):
        """ The name of the country where the well the sample comes from is located """
        return self.__countryName

    @countryName.setter
    def countryName(self, val):
        self.__countryName = val
        
    @property
    def fieldName(self):
        """ The field name for the well where the sample came from """
        return self.__fieldName

    @fieldName.setter
    def fieldName(self, val):
        self.__fieldName = val

    @property 
    def operatorName(self):
        """ The operator name from the well the sample comes from """
        return self.__operatorName

    @operatorName.setter
    def operatorName(self, val):
        self.__operatorName = val

    def composeAbstractElements(self):
        """ Returns a list of strings, each representing a line in the final abstract value """
        ret = super().composeAbstractElements()

        #tack our abstract elements on to the list.
        self.addAbstractElement(self.apiNumber, "API Number", ret)
        self.addAbstractElement(self.leaseName, "Lease Name", ret)
        self.addAbstractElement(self.wellNumber, "Well Number", ret)
        self.addAbstractElement(self.countyName, "County Name", ret)
        self.addAbstractElement(self.stateName, "State Name", ret)
        self.addAbstractElement(self.countryName, "Country Name", ret)
        self.addAbstractElement(self.fieldName, "Field Name", ret)
        self.addAbstractElement(self.operatorName, "Operator Name", ret)

        return ret

File: LibTXNDCPy/LibTXReSciCollCSharp/RSCSampleFromOGWell.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LibTXReSciCollCSharp
{
    /// <summary>
    /// Represents a sample that comes from an oil and gas well.  This class is meant to be an intermediate class that you'd extend with some other type, such as a core, or cuttings.
    /// </summary>
    public class RSCSampleFromOGWell : RSCSample
    {
        /// <summary>
        /// The API number for the oilwell that this sample is derived from
        /// </summary>
        public string? ApiNumber { get; set; }

        /// <summary>
        /// The lease name of the oil or gas well that this sample is drived from
        /// </summary>
        public string? LeaseName { get; set; }

        /// <summary>
        /// Well number of the oil or gas well that this sample is derived from.
        /// </summary>
        public string? WellNumber { get; set; }

        /// <summary>
        /// The name of the state/province/other administrative division where the well is located.
        /// </summary>
        public string? StateName { get; set; }

        /// <summary>
        /// The name of the county/parish where the well is located.
        /// </summary>
        public string? CountyName { get; set; }

        /// <summary>
        /// The name of the country where the well is located.
        /// </summary>
        public string? CountryName { get; set; }

        /// <summary>
        /// The name of the field where the well is located.
        /// </summary>
        public string? FieldName { get; set; }

        /// <summary>
        /// The name of the operator of the well
        /// </summary>
        public string OperatorName { get; set; }

        public RSCSampleFromOGWell() { }

        protected override List<string> ComposeAbstractElements()
        {

            List<string> elements = base.ComposeAbstractElements();

            if (ApiNumber != null)
            {
                AddAbstractElement(ApiNumber, "API Number", elements);
            }

            if(LeaseName != null)
            {
                AddAbstractElement(LeaseName, "Lease Name", elements);
            }

            if(WellNumber != null)
            {
                AddAbstractElement(WellNumber, "Well Number", elements);
            }

            if(StateName != null)
            {
                AddAbstractElement(StateName, "State Name", elements);
            }

            if(CountyName != null)
            {
                AddAbstractElement(CountyName, "CountyName", elements);
            }

            if(CountryName != null)
            {
                AddAbstractElement(CountryName, "Country Name", elements);
            }

            if (FieldName != null)
            {
                AddAbstractElement(FieldName, "Field Name", elements);
            }

            if (OperatorName != null)
            {
                AddAbstractElement(OperatorName, "Operator Name", elements);
            }

            return elements;
        }
    }
}
